Makefile

```diff
@@ -185,121 +199,393 @@ test-integration: test-integration-aws test-integration-clickhouse test-integrat
+	 -e KAFKA_SSL_KEY_PASSWORD=NOPASS -e KAFKA_SSL_ENDPOINT_IDENTIFICATION_ALGORITHM=none \
+	 -e KAFKA_OPTS="-Djava.security.auth.login.config=/etc/kafka/kafka_server_jaas.conf" \
+	 -e KAFKA_INTER_BROKER_LISTENER_NAME=SASL_PLAINTEXT -e KAFKA_SASL_ENABLED_MECHANISMS=PLAIN \
+	 -e KAFKA_SASL_MECHANISM_INTER_BROKER_PROTOCOL=PLAIN -v $(PWD)/tests/data/localhost.p12:/certs/localhost.p12:ro \
+	 -v $(PWD)/tests/data/kafka_server_jaas.conf:/etc/kafka/kafka_server_jaas.conf --name kafka wurstmeister/kafka
+endif
+
+stop-integration-kafka:
+	$(CONTAINER_TOOL) rm --force kafka zookeeper 2>/dev/null; true
+ifeq ($(CONTAINER_TOOL),podman)
+	$(CONTAINER_TOOL) $(CONTAINER_ENCLOSURE) stop --name=test-integration-kafka 2>/dev/null; true
+	$(CONTAINER_TOOL) $(CONTAINER_ENCLOSURE) rm --force --name test-integration-kafka 2>/dev/null; true
+else
+	$(CONTAINER_TOOL) $(CONTAINER_ENCLOSURE) rm test-integration-kafka 2>/dev/null; true
 endif
 
 test-integration-kafka: ## Runs Kafka integration tests
 ifeq ($(AUTOSPAWN), true)
-	${MAYBE_ENVIRONMENT_EXEC} $(CONTAINER_TOOL)-compose up -d dependencies-kafka
-	sleep 5 # Many services are very lazy... Give them a sec...
+	$(MAKE) -k stop-integration-kafka \
+    ; rc=$$? \
+	$(MAKE) start-integration-kafka
+	sleep 10 # Many services are very slow... Give them a sec..
 endif
 	${MAYBE_ENVIRONMENT_EXEC} cargo test --no-default-features --features "kafka-integration-tests rdkafka-plain" --lib ::kafka:: -- --nocapture
 ifeq ($(AUTODESPAWN), true)
-	${MAYBE_ENVIRONMENT_EXEC} $(CONTAINER_TOOL)-compose stop
+	$(MAKE) -k stop-integration-kafka
+endif
+
+start-integration-loki:
+ifeq ($(CONTAINER_TOOL),podman)
+	$(CONTAINER_TOOL) $(CONTAINER_ENCLOSURE) create --replace --name test-integration-loki -p 3100:3100
+	$(CONTAINER_TOOL) run -d --$(CONTAINER_ENCLOSURE)=test-integration-loki -v $(PWD)/tests/data:/etc/loki \
+	 --name loki grafana/loki:master -config.file=/etc/loki/loki-config.yaml
+else
+	$(CONTAINER_TOOL) $(CONTAINER_ENCLOSURE) create test-integration-loki
+	$(CONTAINER_TOOL) run -d --$(CONTAINER_ENCLOSURE)=test-integration-loki -p 3100:3100 -v $(PWD)/tests/data:/etc/loki \
+	 --name loki grafana/loki:master -config.file=/etc/loki/loki-config.yaml
+endif
+
+stop-integration-loki:
+	$(CONTAINER_TOOL) rm --force loki 2>/dev/null; true
+ifeq ($(CONTAINER_TOOL),podman)
+	$(CONTAINER_TOOL) $(CONTAINER_ENCLOSURE) stop --name=test-integration-loki 2>/dev/null; true
+	$(CONTAINER_TOOL) $(CONTAINER_ENCLOSURE) rm --force --name test-integration-loki 2>/dev/null; true
+else
+	$(CONTAINER_TOOL) $(CONTAINER_ENCLOSURE) rm test-integration-loki 2>/dev/null; true
 endif
 
 test-integration-loki: ## Runs Loki integration tests
 ifeq ($(AUTOSPAWN), true)
-	${MAYBE_ENVIRONMENT_EXEC} $(CONTAINER_TOOL)-compose up -d dependencies-loki
-	sleep 5 # Many services are very lazy... Give them a sec...
+	$(MAKE) -k stop-integration-loki \
+    ; rc=$$? \
+	$(MAKE) start-integration-loki
+	sleep 10 # Many services are very slow... Give them a sec..
```

Comment:
No luck on some sort of a healthcheck huh? :( I wonder if we can come up with something clever in the future.

Reply:
Yeah a future iteration.